Libraries/LibWeb/CSS/StyleValues/CalculatedStyleValue.cpp

```diff
@@ -1890,7 +1890,7 @@ Optional<Length> CalculatedStyleValue::resolve_length_percentage(Layout::Node co
 Optional<Length> CalculatedStyleValue::resolve_length_percentage(Length::ResolutionContext const& resolution_context, Length const& percentage_basis) const
 {
     auto result = m_calculation->resolve(resolution_context, percentage_basis);
-    if (result.type().has_value() && result.type()->matches_length())
+    if (result.type().has_value() && (result.type()->matches_length() || result.type()->matches_percentage()))
         return Length::make_px(CSSPixels { result.value() });
```

Comment:
This isn't correct. If a percentage gets returned (eg, `35%`) it'll return that number as a length (`35px`).

A more correct thing to do would be to return the percentage applied to the `percentage_basis` in that case.

The question is why this isn't getting converted to a length earlier on. `NumericCalculationNode::resolve()` should be doing that.

Reply:
From my analysis, I believe it would actually return the correct resolved pixel value (as `percentage_basis` was being called on it  in `NumericCalculationNode::resolve`), the issue was purely that it was being reported as the wrong type, and I do think that my original patch, did _fix_ the specific issue, but agree that it didn't fix the underlying problem.